moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp

```diff
@@ -335,14 +337,32 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
     num_planning_attempts_ = num_planning_attempts;
   }
 
-  void setMaxVelocityScalingFactor(double max_velocity_scaling_factor)
+  void setMaxVelocityScalingFactor(double factor)
   {
-    max_velocity_scaling_factor_ = max_velocity_scaling_factor;
+    setMaxScalingFactor(max_velocity_scaling_factor_, factor, "velocity_scaling_factor", 0.1);
   }
 
-  void setMaxAccelerationScalingFactor(double max_acceleration_scaling_factor)
+  void setMaxAccelerationScalingFactor(double factor)
   {
-    max_acceleration_scaling_factor_ = max_acceleration_scaling_factor;
+    setMaxScalingFactor(max_acceleration_scaling_factor_, factor, "acceleration_scaling_factor", 0.1);
+  }
+
+  void setMaxScalingFactor(double& variable, const double target, const char* name, double fallback)
```

Comment:
This method looks quite ugly to me, especially `name` which has to be a suffix to an incomplete ROS parameter name later on. (you could even drop the `_scaling_factor` the way you're doing it now).

For me, this is clearly not worth avoiding 3 lines of code duplication.

Reply:
I don't get what is ugly about this method. This is 16 lines of code, which would be duplicated.
Code duplication is always error-prone to be not updated in sync in the future.

Reply:
I think I'm with @rhaschke on this one, but I made the variable names longer to spite him 😎 (and because I think it's clearer).

Reply:
I'm fine with this :smile:

Reply:
Ah, I didn't see this was your code @rhaschke . Of course you would argue for more code obscurity over easier-to-understand code :laughing: .
Anyway, it's just a nitpick.